### ISSUE TITLE
Array of dictionaries

### ISSUE THREAD

**Issue by albertolerda** [dyne/Zenroom#288]
If I have an array of dictionaries in the data and I declare it as 'string array' or 'string dictionary' no error is thrown.
To reproduce the problem, go to `apiroom.net`, use as data
```
{
	"query_result": [
		{
			"value": 300,
			"color": "blue"
		},
		{
			"value": "1300",
			"color": "yellow"
		}
	]
}
```
and with the script
```
Given I have a 'string dictionary' named 'query_result'
Then print data
```

**Comment by andrea-dintino** [comment]
I found that using 'string array' works, the Zencode:

```
Given I have a 'string array' named 'array'
Given I have a 'number' named 'number'
Given I have a 'string' named 'string'
Given I have a 'string dictionary' named 'string dictionary'

When I insert 'number' in 'array'
When I insert 'string' in 'array'
When I insert 'string dictionary' in 'array'

Then print data

```

with data:

```
{
	"array": [
		{
			"name": "mario",
			"age": 2,
			"timestamp": 3
		},
		{
			"name": "luigi",
			"age": 5,
			"timestamp": 2
		}
	],
	"number": 128,
	"string": "my sexy string",
	"string dictionary": {
		"ip": "0.0.0.1",
		"pubkey": "abcdesfdsdsdfd",
		"ciccio": "pesce"
	}
}
```

@jaromil maybe we can create 'dictionary array' as an alias of 'string array'? 

Produces the output: 

```
{
  "array": [
    {
      "age": 2,
      "name": "mario",
      "timestamp": 3
    },
    {
      "age": 5,
      "name": "luigi",
      "timestamp": 2
    },
    128,
    "my sexy string",
    {
      "ciccio": "pesce",
      "ip": "0.0.0.1",
      "pubkey": "abcdesfdsdsdfd"
    }
  ]
}
```

**Comment by jaromil** [comment]
The name array or dictionary must be in second position to the definition of encoding when declared.

The definitions array or dictionary refer to the 1st level of depth of the object. So it is correct to use '*encoding* array'

**Comment by matteo-cristino** [comment]
fixed in #788
